dns/zone.py

```diff
@@ -500,16 +501,27 @@ def to_file(self, f, sorted=True, relativize=True, nl=None):
 
         str_type = string_types
 
-        if nl is None:
-            opts = 'wb'
-        else:
-            opts = 'wb'
-
         if isinstance(f, str_type):
-            f = open(f, opts)
+            f = open(f, 'wb')
             want_close = True
         else:
             want_close = False
+
+        # must be in this way, f.encoding may contain None, or even attribute
+        # may not be there
+        file_enc = getattr(f, 'encoding', None)
+        if file_enc is None:
+            file_enc = 'utf-8'
```

Comment:
ASCII characters (at least first 128) in UTF-8 are encoded in the same way.

We cannot drop binary mode support because it violates backward compatibility and existing scripts may stop working.

I don't want to use f.mode, I used it in my first PR, and I regreted it. As file descriptors there are used not just files but some binarystreams even in dnspython code (IIRC, but tests failed). Those dont have mode attribute

It is my bad, I tested it with stdout and it has 'encoding' attribute